src/main/java/com/sendgrid/SendGrid.java

```diff
@@ -1,117 +1,143 @@
 package com.sendgrid;
 
+import org.apache.http.impl.client.CloseableHttpClient;
+
 import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
 
 /**
-  * Class SendGrid allows for quick and easy access to the SendGrid API.
-  */
+ * Class SendGrid allows for quick and easy access to the SendGrid API.
+ */
 public class SendGrid {
-  private static final String VERSION = "3.0.0";
-  private static final String USER_AGENT = "sendgrid/" + VERSION + ";java";
-
-  private String apiKey;
-  private String host;
-  private String version;
-  private Client client;
-  private Map<String,String> requestHeaders;
-
-  /**
-    * @param apiKey is your SendGrid API Key: https://app.sendgrid.com/settings/api_keys
-    */
-  public SendGrid(String apiKey) {
-    this.client = new Client();
-    initializeSendGrid(apiKey);
-  }
-
-  /**
-    * @param apiKey is your SendGrid API Key: https://app.sendgrid.com/settings/api_keys
-    * @param test is true if you are unit testing
-    */
-  public SendGrid(String apiKey, Boolean test) {
-    this.client = new Client(test);
-    initializeSendGrid(apiKey);
-  }
-
-  /**
-   * @param apiKey is your SendGrid API Key: https://app.sendgrid.com/settings/api_keys
-   * @param client the Client to use (allows to customize its configuration)
-   */
-  public SendGrid(String apiKey, Client client) {
-    this.client = client;
-    initializeSendGrid(apiKey);
-  }
-
-  public void initializeSendGrid(String apiKey) {
```

Comment:
Please keep this function also, so that we don't have a breaking change. Generally, please ensure all public functions remain and just add a comment that notes that we will deprecate for now.